### DISCUSSION TITLE
Feature Request: let EarTrumpet save it's settings

### DISCUSSION THREAD

**Discussion by Droyk** [File-New-Project/EarTrumpet#649] [Ideas]
Good evening Developers,
I want a feature from you guys to make my life a little easier let EarTrumpet save it's settings I don't know why you guys didn't implement this yet but plz add it so I can live my life a little easier 

**Everytime I shutdown or restart or signout of my computer ear trumpet forgets everything plz do something about it!!**

**Comment by riverar** [comment]
This is tough because Windows itself doesn't support it. Will investigate as this is a common request.

**Reply by pHROZEN-gHOST** [reply]
Please don't compare your product to Microsoft's OS. It is better than this.

**Comment by Droyk** [comment]
>Will investigate

really 😞😢 I have been hearing that word for a while... and it only seem to me as an excuse as I said on my previous issue it's been 6 months since last ear trumpet update and there is doesn't seem to be any progress on ear trumpet at all. 

tell me how hard it is to implement a saving feature I have seen many UWP apps that saves my history, passwords, etc and you can't implement a single saving feature!!

**Comment by riverar** [comment]
Funny.

Just to be clear -- Are you talking about saving volume levels per app?

**Comment by Droyk** [comment]
>Funny.

No it isn't I like ear trumpet but I want it to have some good features.

>Are you talking about saving volume levels per app?

yup 😤>:(

**Comment by Droyk** [comment]
you didn't reply me @riverar plz tell me when you guys going to work on the next update to add this feature

**Comment by riverar** [comment]
I'm not convinced this feature makes sense to implement. Can you explain in detail why you want this? Do you often start apps that you want at a specific volume? Why?

We talked about how we would implement this feature, it's not trivial but can be done.

**Comment by Droyk** [comment]
>I'm not convinced this feature makes sense to implement.

@riverar now it seems like you are trolling with me and now you are making yourself really stupid.

>I'm not convinced this feature makes sense to implement.

really ask any sensible person they will tell you really easily why this feature is important. 

I will give you a simple answer it make's my life really easy doesn't let me waste my time on muting some apps sounds and changing some apps sound level everytime I restart my computer.

Okay let me give you the same question why do you use bookmark feature in your browser? it doesn't make any sense to bookmark something when a user can easily type it on addressbar. Can you explain in detail why do you use it? Do you often go to those sites if yes then why can't you just pin them? Why? Why? Why?

the answer is simple a user want's to open them without wasting their time on writing the whole URL in the address bar that's the reason they created a bookmark feature that let them open the webpage effortlessly without wasting their time on learning the whole url, writing the url.

Here is another question for you Why do use saving feature in any application at all you just have to remember it right can't you just do that Can you explain in detail why do use a pesky saving feature at all?  Why? Why? Why?

**Comment by riverar** [comment]
What apps are you muting **after a restart**? What apps are you changing the volume of **after a restart**? I'm trying to understand the way you use EarTrumpet.

Keep in mind Windows does not guarantee a start-up order for UWP apps. The apps that you want to restore volume settings for could start before EarTrumpet. This is just one of the many facets of the problem that you may not have considered. It's a complex issue.

**Comment by ghost** [comment]
I think it's the same issue as #23 #31 #88 

For example, in #88, there is no way to mute the audio in the game itself, and it start pretty loudly.

The boot order can be a limitation, but I think it's still beter if the audio is on for a few seconds for an app, then not having this feature at all.

I also miss this feature, because there are some system sounds (like notification sounds) that you can't mute in the old mixer, but you can mute in EarTrumpet, but the app keeps forgetting it on every restart.

**Comment by riverar** [comment]
Thanks @stevek123. Am I missing something?

Solitaire has audio controls here.
![image](https://cloud.githubusercontent.com/assets/475132/21827425/5d70edf8-d740-11e6-9802-15deb2fe2734.png)

And various system sounds, like the one tied to Notifications, can be turned off via 
![image](https://cloud.githubusercontent.com/assets/475132/21827473/978bc3fa-d740-11e6-9da6-cb2efed84235.png)

**Comment by Droyk** [comment]
>What apps are you muting after a restart? What apps are you changing the volume of after a restart?

As @stevek123 said Default one is windows notification ear trumpet tooltip says it is system sound and shell notification sound and then come's some of my media players sound levels

>Keep in mind Windows does not guarantee a start-up order for UWP apps. This is just one of the many facets of the problem that you may not have considered. It's a complex issue.

you just said above that "it's not trivial but can be done." so plz do it I understand the start up order issue & as @steve123 said "but I think it's still better if the audio is on for a few seconds for an app, then not having this feature at all."

@riverar 
yes you can disable those notification sounds but sometimes I want to hear notification sound so I just want a simple toggle through ear trumpet. On 60% of cases I don't want to hear notification sound but on those 40% of cases I want to enable sound via ear trumpet.

**Reply by KoopaTroopa21** [reply]
so why dont you use DND on windows?

**Comment by ghost** [comment]
I don't remember having that option in the game before, I think they have updated it since then (or I was the one who wasn't able to find it :D Now it's definitely there for me too, so thx for the info). 

About the notifications: You are right, it's a workaround (if you want to turn it off completely), but not necessarily a solution.

**Comment by Droyk** [comment]
Yup it is just a workaround not a permanent solution.

**Comment by Droyk** [comment]
@riverar

**Comment by phred73** [comment]
I'd like to request this feature also. As for now, when I start an app it starts with 100% volume. If note possible to save per app maybe setting a default start volume instead (not 100%) of new apps?

**Comment by riverar** [comment]
Windows 10 Creator's Update supports saving app volumes now, so this isn't a feature we need to implement. I tested on 15031.rs2_release.

**Comment by moraks** [comment]
ET 1.4.4.0, win 10  v15063. Volume Level for " Windows Shell Experience Host" not saved. I'm install EarTrumpet special for control new system notification volume level. But it not work... I'm setup 5% of Volume for " Windows Shell Experience Host" , after 5 min it 100% again - terrify!

**Comment by ghost** [comment]
@riverar I'm experiencing the same thing as @moraks. It won't save the mute

**Comment by moraks** [comment]
The problem is also relevant for the new version. "Windows Shell Experience Host" always returns to 100%

**Comment by minimashor** [comment]
@riverar In response to your inquiry about the way I use EarTrumpet, I use it as a mixing board like in a recording studio. I would really appreciate having the ability to set a volume level for each application that would persist after a restart. Not every program has the ability to adjust volume levels, and I would really like to have all of my level sliders in the same place. I would love to help and answer any questions about my usage patterns.
Thanks,
Adam

**Comment by Striferx84** [comment]
Reporting the same issue. I installed latest version of EarTrumpet a couple days ago, to help manage audio levels.

I use a headset, and having my audio levels boost to 100% is not ideal. So I keep the volume of my headset device master volume set at 100%.

I then adjust the volume settings for each individual application, as some applications I want only on 5% volume, or lower; such as notification and alert sounds. Other audio application sounds I keep at 25%, or whatever is comfortable.

As a user mentioned above among several others, once your PC auto-locks/signs out/restarts, all the audio levels reset to the max set "Master volume level" of the device. This is unacceptable, as it is a problem for headset users, and is potentially harmful.

I'm using Windows 10 Home, version 1803, OS Build 17134.112

Is there any update or fix to this problem?

**Comment by moraks** [comment]
@Striferx84  "volume set at 100% for some WinApp"  is a windows bug, but microsoft persistently ignore this... Please, write new bug's report to Microsoft, may be someday they will notice it. I wrote about this problem three times already😂

**Comment by Skcyte** [comment]
I can relate to Striferx84's problem. Everytime there's an app update whether it's windows app or other app it will reset to max volume. I hope there's a solution to this, being to able save the app name instead reset every newer executable is a life saver.

**Comment by ScientificNerd** [comment]
I am not having problems with saving levels on reboot, bar one.  Windows Shell Experience Host always comes up at 100%.  This blasts me out of my seat when a Windows Notification pops up.  Your app is great, but being able to maintain levels of the Shell Experience Host on boot/reboot would be invaluable for me.  Especially since the icon doesn't show up anywhere until it blasts me out of my seat, and never shows up on the Windows mixer, which is why I found EarTrumpet.

If this is strictly a windows only issue, and there's nothing you can do about it, I totally understand.  MS doesn't appear to respond to issues very much though, so I don't expect it will ever be fixed.

**Comment by gaia** [comment]
I'd pay for this feature.

I don't want chat clients (like WhatsApp) or the windows alert beep to constantly ring at the same volume as the music being played. Neither in the same session or even across reboots. I dont ever need these beeps at the same volume as my music.

PS: this info might help https://superuser.com/questions/384472/how-to-make-windows-7s-volume-mixer-save-individual-volumes-again

**Comment by Rocketboy235** [comment]
I also would like to see this be implemented as well. I currently have my volume up pretty high and set the rest of my programs to low volume except for my music player. I also have some alarms on my computer so it gets a bit annoying when they go off really loud since EarTrumpet does not save the Windows Shell Experience volume. Of course, I could just rethink my strategy on handling volume for all my programs and music but prefer not to since I have other speakers around me not independent of my computer's volume so I have to match everything up. Also, unfortunately, there is no way to modify the volume specifically for my alarm app and I don't want to have to look for another alarm app because of this since the alarm app I like is pretty good (with the exception you cannot control the volume individually)

**Comment by bluearcher-bc** [comment]
> I also would like to see this be implemented as well. I currently have my volume up pretty high and set the rest of my programs to low volume except for my music player. I also have some alarms on my computer so it gets a bit annoying when they go off really loud since EarTrumpet does not save the Windows Shell Experience volume. Of course, I could just rethink my strategy on handling volume for all my programs and music but prefer not to since I have other speakers around me not independent of my computer's volume so I have to match everything up. Also, unfortunately, there is no way to modify the volume specifically for my alarm app and I don't want to have to look for another alarm app because of this since the alarm app I like is pretty good (with the exception you cannot control the volume individually)

I am with you, but I think the WIndows Shell Experience issue is a Windows one not an EarTrumpet one. 

I really hate it though.

**Comment by ender4** [comment]
I would also like to request this feature. I found this app by searching for the Windows Shell Experience Host volume issue.

**Comment by tjsepka** [comment]
Same issue / problem here.  I also see it with Firefox (FF) tabs.  With FF, usually only one control returns to its previous level and all others are set to Volume Mixer's Device (a.k.a., "output volume") slider level.  The system notification level control (**not the System Sounds application slider on the Volume Mixer!!!**) _always_ (as far as I can remember), is set to the Volume Mixer's Device's level.

It's not unusual in pro-audio applications to set the power amplifier's level control (if it has one at all!) to maximum and control all audio levels from the mixing board.  I'm an Electrical Engineer and used to design pro-audio equipment used in churches, school auditoriums, stadiums, theaters, and other large venues, and our power amp had screwdriver-adjust trim pots, only because we didn't always have knowledgeable pro's using Pro equipment, and they were used to having volume pots.  So setting Windows device level to maximum had been my approach when using Window's Mixer.  Bad news when you drop in a CD and Window's notification blasts at you asking you what to do with it...

Anyway, I don't have a solution, but I'm looking at NirSoft's "SoundVolumeView" and "AppAudioConfig" capabilities (when I have time) and would ask others with this issue have a look, too.

I always welcome helpful and useful comments and suggestions!

**Comment by riverar** [comment]
Hey everyone, this is supposed to be supported by Windows but agree, it's broken. I'm working with Microsoft on this. In the meantime, we also have an EarTrumpet add-on in the works that will give you the ability to set volumes per app when they start, end, etc. Bear with us, we're close. Thanks!

**Comment by tjsepka** [comment]
Thank you for the update, Rafael.  That's great news!

Once that's fixed, it would be nice to have EarTrumpet implement the classic Windows mixer layout as an option, one which adds minimize and restore functions, as well as being fully resizable.  I use a dual monitor setup and dock Windows mixer in the lower right hand corner of my secondary right hand monitor, with my left hand monitor being my primary.

**Comment by Rocketboy235** [comment]
Thanks Rafael! Really appreciate you following up and also putting the hard work into making this.

**Comment by riverar** [comment]
> Once that's fixed, it would be nice to have EarTrumpet implement the classic Windows mixer layout as an option, [...]

Heads up, we do have a volume mixer very similar to the classic one. Or were you looking for something else?

![image](https://user-images.githubusercontent.com/475132/53304954-375df180-3830-11e9-866f-d2bc2f18ce8f.png)

**Comment by tjsepka** [comment]
For the life of me, I can't find the GUI you show above!  Can you advise me as to what I'm doing wrong?

**Comment by riverar** [comment]
@tjsepka If you right-click the EarTrumpet icon in the notification area, you get this context menu 
![image](https://user-images.githubusercontent.com/475132/53306765-43ed4480-3846-11e9-8c55-05c6d8b0923c.png)

Click EarTrumpet Volume Mixer 👍

If you don't see this, it sounds like you're on an old version of EarTrumpet. Make sure you grab the latest from the [Microsoft Store](https://www.microsoft.com/en-us/p/eartrumpet/9nblggh516xp)

**Comment by tjsepka** [comment]
Got it!  I ran across that screen, but didn't look closely enough at it.  My impressions are:

1.) The screen cannot be minimized;
2.) The screen cannot be resized.  It takes up considerable area of my monitor;
3.) The peak indicators only show significant deflection (levels) when the control is turned up quite high.  This might be occurring if the transfer function being used is linear rather than logarithmic (also known as "audio tapered"}.
4.) It would be nice if each channel's level indicator had its own color.
5.) It appears that I can't control the slider position with my mouse wheel like I can with Windows mixer.
6.) It would be nice to have the option to position the slider controls vertically.

**Comment by factormystic** [comment]
> Windows 10 Creator's Update supports saving app volumes now, so this isn't a feature we need to implement. I tested on 15031.rs2_release.

Is this a standard windows feature, or a new API exposed to apps to save/restore their own volume levels? I've noticed that closing/reopening a uwp app will cause its volume to reset to 100%, but I can't tell if this is what is being discussed in this thread (not really about rebooting, and not sure if it's an eartrumpet thing or not).

**Comment by riverar** [comment]
@factormystic It's supposed to Just Work ™️ with no exposed API. But we haven't seen it work at all these days, suspect it's broken. Please also file a Feedback Hub issue (in the audio category) and link me so I have the ammunition to push back on Microsoft in our thread. Thanks!

**Comment by Testi89** [comment]
Saving the volume setting would be great. I use TuneIn Radio app with a headset, and I always have to correct from 100 to 1 not to get blown away. 
I use Windows 10, 1709, build 16299.1087

**Comment by PaddyMcQ** [comment]
This was a painful thread to read.  Anyway, I installed Ear Trumpet specifically to normalize volume levels between Microsoft Edge browsing sessions and my local apps.  Sampling music on various websites using Edge, the volume is always twice as loud as my local music player.  I cannot adjust my master volume and/or player volume to compensate.  Ear Trumpet works great and fixes this but... the volume level set in Ear Trumpet are not saved when I reboot.  Thanks.

**Comment by sagegriesfox** [comment]
Having EarTrumpet create a Persistent App volume is the SINGLE MOST VALUABLE feature I feel you could ever add.  Indeed it was the reason I installed EarTrumpet !  I have a whole-home music system and home automation center than controls our background music 24/7 as well as home environmental conditions.
The system (running Win 10) regularly reboots itself to manage software updates as well as general house-cleaning.  Every time it reboots all of the dings, dongs, beeps, and bleeps that the system and various App create (to the annoyance of home music listeners) become loud again and I have to manually reset them to a level of 5% so they can be heard but not heard.

How hard can it be, no matter what order Apps are loaded, to store settings in the Registry and then examine and correct things as needed 30 seconds after EarTrumpet starts up and things have stabilized?

PLEASE, PLEASE, PLEASE, PLEASE, PLEASE do this as it is badly needed and you will be the ONLY one in the whole world that solves this issue!

 [:-)>

**Comment by tjsepka** [comment]
I agree & I feel your pain
!

**Comment by mgutt** [comment]
I thought EarTrumpet could solve my problem but with this "bug" its not a real fix.

Finally I solved it by reducing the volume of all Windows Notification WAV-files. If you speak German or able to use a translator you can follow this How-To:
https://www.maxrev.de/plex-ist-zu-leise-oder-windows-benachrichtigungen-sind-zu-laut-t394161.htm

**Comment by drawbars** [comment]
Let me add another example of why saving the volume for applications is closer to a deal-breaker than a nice-to-have ...

I'm a musician.  I tend to keep my Windows system prompts, browser, etc. at a moderate level; let's call it 30 (out of 100) for the sake of discussion.  I have other apps like Sonar and Nero Wave Editor that I use to create 'product,' and when I'm using them, I probably want them at 50-60 while I'm working, and occasionally at 100 to audition the final results.  And then I have background music which is often on when I'm doing other work, which I tend to like at slightly higher level than system sounds -- maybe, 40?  In all those examples, I still want my 'other' sounds to stay at a moderate level:  there's nothing worse than auditioning a final mix, interrupted by a thundering report of a new email!  So, in my case, 100 is more of an exception than a typical level; having 100 be the starting point when apps open is a bad thing, because 100 = pretty damn loud.  I don't mind occasionally adjusting an app up or down to match a situation, but I don't want to open every other app and find that it's screaming at me.

In my case, mgutt's suggestion of saving off low-volume versions of the system sounds would fix some of the problems, but not all of them.

Regards,
-BW

**Comment by riverar** [comment]
Hey folks, want to provide a quick update -- we agree. And we built this feature. We're just jumping through a bunch of administrative hoops to get our EarTrumpet add-on framework in a position where it's ready for use. My apologies for taking so dang long but we're close! We'll keep you all in the loop.

**Comment by martonx** [comment]
Hi, do you have any news? I use Eartrumpet, because of I have a lot of sound related apps are running in the same time.
It is very disappointing to set all of the sound levels after every PC restart.

**Comment by swilliams-dev** [comment]
@riverar I'd like to see the settings saved off to a config file that it loads on startup. My use case is that I swap between various audio routing configurations using a combination of "app volume and device preferences", voicemeeter, and nircmd's setdefaultsounddevice to handle various audio profiles (recording, broadcasting, streaming to VRChat, streaming to Discord, desktop mode, vr mode, etc). I do this by taskkilling and replacing configs... very hacky but it works for everything but "app volume and device preferences", which EarTrumpet could solve.

Digging into it, but I may have found an alternative mid-post! lol:
https://blog.nirsoft.net/2020/05/29/set-default-audio-device-of-specific-application-from-command-line-on-windows-10/

**Comment by riverar** [comment]
Understood. We rely on Windows to remember audio levels, but despite past assurances by Microsoft that this functionality exists, it doesn't seem to work at all.

Please file a bug in Feedback Hub under [Devices and Drivers] - [Audio and sound]. You can cut and paste from this template (but verify the steps apply to you):

```
Summary: Windows does not remember audio levels for application ____

Explain in more detail:
When I start my application, its audio level is reset to 100%,
requiring me to change it again.

Repro steps:
  1. Start application ____
  2. Set audio level to 25%
  3. Close application and restart machine
  4. Start application ____
  5. Observe audio level has reset to 100%

See also: EarTrumpet/issues/98
```

Then paste a link here and I'll forward it directly to the audio team.

In the meantime, we have an EarTrumpet add-on built for this and will ship it soon.

**Comment by Ownsin** [comment]
Any updates on this? It has been 3 months and no updates on shipping this feature out...

**Comment by mgutt** [comment]
> Finally I solved it by reducing the volume of all Windows Notification WAV-files.

Only as feedback. I'm still using this workaround (since one year) and it works perfectly.

**Reply by T3OT3TT3** [reply]
hi can u tell me more about the workaround?
Thanks

**Comment by stressatoo** [comment]
Could we get some more information and/or updates? @riverar
I'm using discord and there's no way to reduce the mute/unmute volume since the assets are stored in cloud, and it's really loud for me.. so everytime I restart my PC I need to right click discord on eartrumpet and lower one of the 2 sliders.
![image](https://user-images.githubusercontent.com/60473020/125190161-ef429c80-e23b-11eb-9a90-73660e65fa68.png)

**Reply by Ownsin** [reply]
I agree. We need Eartrumpet to save sound changes. This feature is a must...

**Reply by riverar** [reply]
Have you tried our dev builds? (Click the text link on our homepage @ https://eartrumpet.app.) They come with EarTrumpet Actions that can be used to set volume levels on a per-app basis when they're launched.

**Reply by Ownsin** [reply]
> Have you tried our dev builds? (Click the text link on our homepage @ https://eartrumpet.app.) They come with EarTrumpet Actions that can be used to set volume levels on a per-app basis when they're launched.

I tried it, and it still doesn't save the sound settings for my programs.

**Reply by stressatoo** [reply]
> Have you tried our dev builds? (Click the text link on our homepage @ https://eartrumpet.app.) They come with EarTrumpet Actions that can be used to set volume levels on a per-app basis when they're launched.

  I can't! That leads to an error because your link isn't working. I analyzed the link and that leads to [this .appinstaller file](https://install.eartrumpet.app/dev/EarTrumpet.Package.appinstaller).. that doesn't work! So that might be why.

![image](https://user-images.githubusercontent.com/60473020/126067127-ab903416-4c61-4526-9d1c-cb9346624194.png)

**Reply by aadityaverma** [reply]
> 
> 
> Have you tried our dev builds? (Click the text link on our homepage @ https://eartrumpet.app.) They come with EarTrumpet Actions that can be used to set volume levels on a per-app basis when they're launched.

Will the volume levels per app be remembered on a system restart  ?

**Comment by Slegnor** [comment]
Has anyone's suddenly started saving levels correctly in the last few weeks?
I had an issue with VoiceAttack program, volume slider stayed saved at 15%, but the volume always played at 100% until you adjusted the slider.
Now it shows 15% and plays at 15% each launch as I want it to, did somehting get fixed or changed at Microsoft's end?

**Reply by Slegnor** [reply]
And now in November it’s reverted to how it was, the app's slider says 15% volume, but it's blasting out at 100% again

**Reply by naddel91** [reply]
still no fix after all these years?

**Comment by naddel91** [comment]
I love eartrumpet, but I hate to tell it the preferred volume each time I launch a game (again). please fix this.

**Comment by BDtetra** [comment]
This is sad its not fixed, i think it was working in windows 10 but its definitely not working in windows 11 (fresh and clean install).  I need this feature since i do not want to fidget around with volume per app basis, because often times volume controls simply do not exist in apps, resets per update, have no "master volume" option and only "individual" volume sliders.  

This is where I was using EarTrumpet, so I could set all the apps relatively to each other very easily, and forget about it until a new app comes where i need to relatively adjust the volumes again, kind of like a audio mixer.  Without the save feature, EarTrumpet, for me, loses its appeal over something like Xbox Game Bar which serves the same purpose with huge bloat and annoying UI, but actually saves the audio level.  

Saving the Audio Level is absolutely crucial to the function of EarTrumpet as many people use it.

**Reply by riverar** [reply]
Saving audio levels is not a feature of EarTrumpet or Xbox Game Bar. Instead, the OS handles this. I'd be very curious if you can reproduce an app having its levels saved in Xbox Game Bar but not EarTrumpet. If you do have repro steps for that, please file an issue, I'd be happy to investigate/fix if so.

**Reply by BDtetra** [reply]
Sorry, you are correct, I misunderstood how the volume control worked.  I was able to fix the issue after changing to a different sound driver.

**Reply by exitss** [reply]
@BDtetra how would I go about changing my audio driver on win 11 to make it save the volume of each application even after a restart?

**Comment by WilliamK7** [comment]
Hello guys, I find some weird things. For some apps, Windows remembers their volume settings, but there are some exceptions.
You can see in the video I uploaded that Chrome is not able to be remembered, but the other one succeeds.

https://user-images.githubusercontent.com/82304956/212465116-d47bd98d-73b0-48e5-a3f0-d23098add116.mp4

FYI

PS: I really love ET! It's absolutely a fascinating app. Thanks for the developers' hard work.

**Reply by WilliamK7** [reply]
Wait, this is a problem of Chrome, because only it can't be remembered, the other apps all succeed. Sorry for the inconvenience.

**Reply by riverar** [reply]
No problem. I think it's ultimately a Windows bug. It's supposed to remember these things :(

**Reply by davidoffc** [reply]
The funny thing is that prior to the latest emergency reinstall of Windows and all apps on a Dell workstation this (EarTrumpet remembering settings) used to work properly, and it still does on a Dell laptop and another Dell workstation where settings are saved meaning that after reboot EarTrumpet reads the settings (even for Chrome) :)
So where's the catch?

**Reply by riverar** [reply]
EarTrumpet doesn't remember volume settings. That's a Windows feature. We just get numbers from the OS.

When you reinstalled Windows, perhaps you installed newer updates/software that changed the experience you previously had.

**Reply by davidoffc** [reply]
@riverar

> EarTrumpet doesn't remember volume settings. That's a Windows feature. We just get numbers from the OS.
> 
> When you reinstalled Windows, perhaps you installed newer updates/software that changed the experience you previously had.

Oh I believe you, it's just plain odd that some settings are stored while others (chrome's) aren't. Can't figure out where Windows or Chrome stores these settings, is it the registry or somewhere else?

**Comment by Mr0merta** [comment]
I had this issue for so long and finally solved it by using (VolumeLock) which works perfectly so far.

**Reply by WilliamK7** [reply]
Can you pls share a link? Thx a lot😁

**Reply by The-Arcade-Buffett** [reply]
yes, please share a link to the (VolumeLock) program or instructions.

Thanks

**Reply by Slegnor** [reply]
Without posting links to another company's software here, It's literally the top result when you Google "VolumeLock", but it isn't a free program.

**Reply by The-Arcade-Buffett** [reply]
thank you.

**Reply by Droyk** [reply]
Fucker wants money for it a/f 14 days! https://www.actualsolution.com/register/

**Reply by Gravifer** [reply]
You are just a hopeless giant baby weeby dweeb who can only yap and not code, aren't you?

**Reply by riverar** [reply]
Hey all, you're welcome to share links to any software here, as long as it's in good taste. We're all friends just trying to make audio suck less in Windows 👍

**Reply by Gravifer** [reply]
yeah, but i really think you don’t need to be so nice to a pathetic being like this OP @riverar 

I’m from China and you know what we call jerks like these on our forums what? Hand-willows ("伸手党"). And that’s when we have censorship and no respect of open-source in our society what-so-ever, you know, how that sorta stuff.

__Be appropriatly mean, and defend your hard work as you should.__

**Comment by galbaras** [comment]
I'm hoping to restart this discussion with a suggestion.

It seems that right now, EarTrumpet simply provides a better interface to what the Windows Audio Mixer already offers and makes changes to the actual Windows volume levels. To get around this, EarTrumpet can also write the selected volume levels into a configuration file and set the appropriately when Windows starts.

Personally, I find that some levels are forgotten when apps are closed, only to be 100% again when they are reopened, so it will be good if EarTrumpet used some sort of Windows trigger when apps start to set volume levels from its configuration file.

**Reply by riverar** [reply]
Thanks @galbaras, try a dev build, which includes EarTrumpet Actions. This will allow you to build up actions that occur when certain triggers fire, such as startup or hotkey press.

https://EarTrumpet.app

**Reply by galbaras** [reply]
Actions can only be exported and imported. How do I use this feature?

**Reply by riverar** [reply]
Click the `+` button to create a new action.

![image](https://github.com/File-New-Project/EarTrumpet/assets/475132/ebc59e84-e4d9-4c18-a02e-d8755f51fc7c)

Then use the controls on the right to add triggers/actions/filters as needed.

![image](https://github.com/File-New-Project/EarTrumpet/assets/475132/99cea67c-de19-4386-aa64-5683087ec7c3)

In this example, when VLC starts playing audio on the default device, its volume is set to 42%.

**Comment by galbaras** [comment]
This is so great! I've been able to add triggers for some programs and they seem to work fine, although I haven't restarted my PC since.

It would be great to have a "catch all" trigger as a default for all apps that aren't defined explicitly. Then, I could set a default volume and only adjust the occasional application. Far less work.

Also, if a program isn't running, it's not possible to add a trigger for it. It's not a big deal over time, but requires exiting the settings and going back into them multiple times, which is a hassle.

**Reply by riverar** [reply]
I'm not at my PC this moment but there should be the equivalent of an "any" in the app list. Will update this when I return if you haven't found it by then. Thanks!

**Reply by Ownsin** [reply]
Please confirm this feature because It would really be great to have an all-app trigger!

**Reply by riverar** [reply]
Doesn't appear we have an "any" app trigger, I'll get that added now.

**Reply by riverar** [reply]
Added and checked in. https://github.com/File-New-Project/EarTrumpet/commit/950ec703ef7427185472efc3dba369f3dfadf601

After the build/deploy is complete, you can restart your build twice and it should automatically update.

![image](https://github.com/File-New-Project/EarTrumpet/assets/475132/765cb465-064e-4fa7-8142-6a326bafe9a8)

**Reply by riverar** [reply]
I just realized you probably want to only set the volume of the newly started app. Let me sleep on that and I'll get you guys another build.

**Reply by galbaras** [reply]
I recently noticed thew "every app" option for triggers and used it. Works like a charm. Well done!

**Reply by dsengar** [reply]
Has this been fixed?

**Comment by Thornskade** [comment]
It doesn't seem to work for me. What does the Process Name field expect? It says "Example: Notepad" and that doesn't really make any sense to me. Does that check notepad.exe so I don't have to type out .exe? Either way I tried putting in a generic name, the name the window of the process displays when hovering over it in the taskbar, the process name including the .exe, I just can't get this to work

**Comment by archenovalis** [comment]
Would absolutely love this feature by default. Every manually adjusted slider could auto create a trigger.
This is especially important because Discord doesn't have a notification volume... and it is SUPER loud. Discord has two sliders, one for notifications and one for everything else. We def need the ability to lock the notification volume low, even if it is only to fix Discords refusal to allow us to adjust notification volume within Discord.

**Reply by mraerozz** [reply]
This is something that has been bothering me for years now too, but I guess this issue is for the most part on discords side. I don't understand as well why there are 2 seperate Audio outputs and you can only adjust 1 of them. For me the problem is that a lot of times the join/leave sounds are just not being played because that audio source is muted somewhere in the background.

**Comment by robinhugy** [comment]
+1
i want to save volume settings too :]
